pgjdbc/src/test/java/org/postgresql/test/jdbc3/Jdbc3CallableStatementTest.java

```diff
@@ -217,11 +218,11 @@ public void testGetObjectDecimal() throws Throwable {
       cstmt.registerOutParameter(3, Types.DECIMAL);
       cstmt.executeUpdate();
       BigDecimal val = (BigDecimal) cstmt.getObject(1);
-      assertTrue(val.compareTo(new BigDecimal("999999999999999.000000000000000")) == 0);
+      assertEquals(new BigDecimal("999999999999999.000000000000000"), val);
```

Comment:
Do you know there's a difference between `BigDecimal#equals` and `BigDecimal#compareTo==0`?
The former treats `2.0 not equal to 2.00` while `compareTo` treats them as equal.

I'm not sure what was the intention here, however this looks like the most suspicious change.

Reply:
I didn't know.
I've reverted usage of eauals